Convert Opcode enums to an enum class

AlexsanderDamaceno opened the pull request:
#### 7dde2cfffbb1c40c0750f69814972daeb119c7ba
<pre>
Convert Opcode enums to an enum class
<a href="https://bugs.webkit.org/show_bug.cgi?id=281111">https://bugs.webkit.org/show_bug.cgi?id=281111</a>

Reviewed by Matthieu Dubet.

Convert Opcode enums to an enum class

* Source/WebCore/xml/XPathParser.cpp:
(WebCore::XPath::Parser::nextTokenInternal):
* Source/WebCore/xml/XPathPredicate.h:
* Source/WebCore/xml/XPathPredicate.cpp:
(WebCore::XPath::NumericOp::evaluate const):
(WebCore::XPath::EqTestOp::compare const):
(WebCore::XPath::LogicalOp::shortCircuitOn const):
(WebCore::XPath::evaluatePredicate):
* Source/WebCore/xml/XPathGrammar.y:
* Source/WebCore/xml/XPathGrammar.cpp:

Canonical link: <a href="https://commits.webkit.org/285024@main">https://commits.webkit.org/285024@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5c1dff72d079e4df7cab710c528ed15c83e3ce50

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/70873 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50285 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23646 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/74977 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22078 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/72989 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58084 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/21896 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56087 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14564 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/73939 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61108 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36536 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/18541 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/20419 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18903 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76692 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15105 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18106 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63823 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15149 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61167 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63769 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15771 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11855 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5506 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46086 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/861 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47158 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48441 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/46900 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on Source/WebCore/xml/XPathPredicate.h:
```diff
@@ -96,7 +96,7 @@ class EqTestOp final : public Expression {
     WTF_MAKE_TZONE_ALLOCATED(LogicalOp);
 public:
-    enum Opcode { OP_And, OP_Or };
+    enum Opcode : uint8_t { And, Or };
```
It could be a `enum class Opcode : bool`

AlexsanderDamaceno replied:
@mdubet  changed.

mdubet replied:
It is still a `uint8_t` ?

AlexsanderDamaceno replied:
@mdubet  sorry, did not pay attention to the bool change, now is right.

AlexsanderDamaceno committed:
Convert Opcode enums to an enum class
https://bugs.webkit.org/show_bug.cgi?id=281111

Reviewed by Matthieu Dubet.

Convert Opcode enums to an enum class

* Source/WebCore/xml/XPathParser.cpp:
(WebCore::XPath::Parser::nextTokenInternal):
* Source/WebCore/xml/XPathPredicate.h:
* Source/WebCore/xml/XPathPredicate.cpp:
(WebCore::XPath::NumericOp::evaluate const):
(WebCore::XPath::EqTestOp::compare const):
(WebCore::XPath::LogicalOp::shortCircuitOn const):
(WebCore::XPath::evaluatePredicate):
* Source/WebCore/xml/XPathGrammar.y:
* Source/WebCore/xml/XPathGrammar.cpp:

Canonical link: https://commits.webkit.org/285024@main
